src/engines/basic-render-engine.ts

```diff
@@ -12,6 +12,25 @@ const checkSafari = () => {
     return ua.includes('safari') ? !ua.includes('chrome') : false;
 };
 
+function createPatternCanvas(blockHeight: number, parentCtx: CanvasRenderingContext2D): CanvasPattern {
+    const patternCanvas = document.createElement('canvas');
+    const patternContext = patternCanvas.getContext('2d');
+    const lineWidth = blockHeight / 1.5;
+    patternCanvas.width = blockHeight + lineWidth;
+    patternCanvas.height = blockHeight;
+    if (!patternContext) {
+        throw new Error("Couldn't create pattern canvas!");
+    }
+    patternContext.fillStyle = 'rgba(255, 255, 255, 0)';
+    patternContext.fillRect(0, 0, patternCanvas.width, patternCanvas.height);
+    patternContext.lineWidth = lineWidth;
+    patternContext.strokeStyle = 'pink';
```

Comment:
I think this color variable should be in node data. You also can provide new API for creating patterns for users, something like a:
```ts
type Settings = {
  patterns: {
    variants: {
      [name: string]: (color: string, style: RenderStyles, patternCanvasContext: CanvasRenderingContext2D) => void
    },
    colors: string[],
  },
}
```
After parsing settings and during the initialization of RenderEngine, you can build a map with all patterns (each combination between colors and variants).

And you can include this pattern as a base pattern. And in each interval object, there should be a `pattern: string` field with the name of the pattern used (from the settings or from the default values) and a `color: string` field.